pkg/topo/options/options.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2020-present Open Networking Foundation <info@opennetworking.org>
+//
+// SPDX-License-Identifier: LicenseRef-ONF-Member-1.0
+
+package options
```

Comment:
I don’t really like having a separate `options' package — `options.WithFoo` makes for a bit less readable code than e.g. `topo.WithFoo` or `client.WithFoo`. This can be put in an `options.go` file in the same package. This also allows the `Option` interface to have a private method so options can’t be applied from outside the package.